tests/tests.cpp

```diff
@@ -564,12 +531,9 @@ TEST_F(EngineBasicTest, TestSortedRestore) {
       ASSERT_EQ(v, value + std::to_string(id));
     }
   };
-  std::vector<std::thread> ts;
-  for (int i = 1; i <= num_threads; i++) {
-    ts.emplace_back(std::thread(ops, i));
-  }
-  for (auto &t : ts)
-    t.join();
+
+  LaunchNThreads(num_threads, ops, 1);
```

Comment:
This.
LaunchNThreads(num_threads, ops, 1);
->
LaunchNThreads(num_threads, ops, 0);
will fail the test. (even if line 490 and 491 are modified)

Reply:
Because id is used to decide what  key and value look like

Reply:
I know that, but why cannot we start from 0? (Already tried modify for loops accordingly and modify string constructors to avoid 0-length string but the test case still fails.)
We should have a vector keep track of the keys and values to be inserted, deleted or updated instead of hand-writing kv-pairs and do copy-paste dirty work.